### PR TITLE
Banner api endpoint

### PR DESCRIPTION
**Pull request by ngambino0192**
This PR adds an api endpoint to fetch featured company data to render in the animated header portion of the app.

Endpoint: `GET /api/markets/promo`

Sample simplified output: array of objects:
```
[
    {
        "symbol": "PLM",
        "companyName": "PolyMet Mining Corp.",
        "calculationPrice": "close",
        "open": 3.3,
        "openTime": 1610238611754,
        "close": 4.4,
        "closeTime": 1620122802444,
        "high": 6.56,
        "low": 3.07,
        "latestPrice": 4.6,
    }
...rest
]
```

A couple of notes:

- We're leveraging a new api for this data. They provide a sandbox env with unlimited api calls. Lmk when you plan on working on this and I can shoot you over the env keys.

- There is a lot more data exposed, but you probably won't need all of it.

- Eventually, I'd like to cache these results somehow, either on the frontend, or maybe in redis to prevent excessive api calls. Left a note as a reminder. Since we are using sandbox api results, this shouldn't be an immediate concern.

### CODE REVIEW

**Review by ngambino0192 — COMMENTED** [review]
Should we add `yarn.lock` to the .gitignore?

**Comment by melsoriano** [comment]
> Should we add `yarn.lock` to the .gitignore?

Lock files should be committed so we both have exactly the same dependency versions. https://stackoverflow.com/a/44904494

**Comment by melsoriano** [comment]
>     * There is a lot more data exposed, but you probably won't need all of it.

I think we should figure out what data we need and only send the required data in the response.

>     * Eventually, I'd like to cache these results somehow, either on the frontend, or maybe in redis to prevent excessive api calls. Left a note as a reminder. Since we are using sandbox api results, this shouldn't be an immediate concern.

Do you know how often these results get updated? Is the banner suppose to have the latest data? We can cache the results on the frontend and only make an api call if the data is updated at specific intervals.

File: backend/app/routes/markets.js [hunk 1]
```diff
@@ -76,4 +77,43 @@ router.get("/quote/:symbol", async (req, res) => {
...
+
+    let collection = [
+      ...gainers,
+      ...losers,
+      ...mostactive,
+      ...volume,
+      ...percent,
+    ];
+
+    let hashmap = {};
```
Comment (melsoriano) [inline backend/app/routes/markets.js:98]: Can we use a more meaningful varibale name for this? 😬

Comment (ngambino0192) [reply backend/app/routes/markets.js]: good point: https://github.com/ngambino0192/stocks/pull/12/commits/e42c0ea98c9517467363d22d3f18af1256b26dce

**Comment by ngambino0192** [comment]
> Do you know how often these results get updated? Is the banner suppose to have the latest data? We can cache the results on the frontend and only make an api call if the data is updated at specific intervals.

Technically, yes it's supposed to have the latest data, but I think we can probably get away with updating it every few hours (maybe 4?). For the production api, there is a limit to 50,000 api calls per month (at the $9/month plan....which I will probably subscribe to).

You think we can address this in a separate task? I think it would be easier to implement once it's hooked up to some front end components.

**Comment by ngambino0192** [comment]
> I think we should figure out what data we need and only send the required data in the response.

I've taken a shot at simplifying this down to what I _think_ we'll need for this feature.

I know this is typically not how we restructure objects, but it looked cleaner that the way we usually do (ie, `const {...symbol} = item`)

This seemed a bit cleaner to me:

```
        uniqueMarketSymbols[symbol] = {
          symbol: item.symbol,
          companyName: item.companyName,
          latestTime: item.latestTime,
          latestPrice: item.latestPrice,
          high: item.high,
          low: item.low,
          change: item.change,
          changePercent: item.changePercent,
          isUSMarketOpen: item.isUSMarketOpen,
        };
```

**Comment by melsoriano** [comment]
> Technically, yes it's supposed to have the latest data, but I think we can probably get away with updating it every few hours (maybe 4?). For the production api, there is a limit to 50,000 api calls per month (at the $9/month plan....which I will probably subscribe to).
> 
> We also don't need to fetch this data if markets are closed.

I see, yeah we can have it fetch every 4 hours and cache the previous data between intervals.  Is there a free tier? $9/month seems like a lot 😬 

> You think we can address this in a separate task? I think it would be easier to implement once it's hooked up to some front end components.

A separate task for all of this works. 👍

**Review by melsoriano — APPROVED** [review]

**Pull request merged**